ml-agents-envs/mlagents_envs/registry/binary_utils.py

```diff
@@ -0,0 +1,198 @@
+        raise NotImplementedError("No extensions found for this platform.")
+    url_hash = "-" + hashlib.md5(url.encode()).hexdigest()
+    path = os.path.join(bin_dir, name + url_hash, "**", extension)
+    candidates = glob.glob(path, recursive=True)
+    if len(candidates) == 0:
+        return None
+    else:
+        return candidates[0]
+
+
+def get_tmp_dir() -> Tuple[str, str]:
+    """
+    Returns the path to the folder containing the downloaded zip files and the extracted
+    binaries. If these folders do not exist, they will be created.
+    :retrun: Tuple containing path to : (zip folder, extracted files folder)
+    """
+    MLAGENTS = "ml-agents-binaries"
+    TMP_FOLDER_NAME = "tmp"
+    BINARY_FOLDER_NAME = "binaries"
+    mla_directory = os.path.join(tempfile.gettempdir(), MLAGENTS)
+    if not os.path.exists(mla_directory):
+        os.makedirs(mla_directory)
+    zip_directory = os.path.join(tempfile.gettempdir(), MLAGENTS, TMP_FOLDER_NAME)
+    if not os.path.exists(zip_directory):
+        os.makedirs(zip_directory)
+    bin_directory = os.path.join(tempfile.gettempdir(), MLAGENTS, BINARY_FOLDER_NAME)
+    if not os.path.exists(bin_directory):
+        os.makedirs(bin_directory)
+    return (zip_directory, bin_directory)
+
+
+def download_and_extract_zip(url: str, name: str) -> None:
+    """
+    Downloads a zip file under a URL, extracts its contents into a folder with the name
+    argument and gives chmod 755 to all the files it contains. Files are downloaded and
+    extracted into special folders in the temp folder of the machine.
+    :param url: The URL of the zip file
+    :param name: The name that will be given to the folder containing the extracted data
+    """
+    zip_dir, bin_dir = get_tmp_dir()
+    url_hash = "-" + hashlib.md5(url.encode()).hexdigest()
+    binary_path = os.path.join(bin_dir, name + url_hash)
+    if os.path.exists(binary_path):
+        shutil.rmtree(binary_path)
+
+    # Download zip
+    try:
+        request = urllib.request.urlopen(url)
+    except urllib.error.HTTPError as e:  # type: ignore
+        e.msg += " " + url
+        raise
+    zip_size = int(request.headers["content-length"])
+    zip_file_path = os.path.join(zip_dir, str(uuid.uuid4()) + ".zip")
+    with open(zip_file_path, "wb") as zip_file:
+        downloaded = 0
+        while True:
+            buffer = request.read(BLOCK_SIZE)
+            if not buffer:
+                # There is nothing more to read
+                break
+            downloaded += len(buffer)
+            zip_file.write(buffer)
+            downloaded_percent = downloaded / zip_size * 100
+            print_progress(f"  Downloading {name}", downloaded_percent)
+        print("")
+
+    # Extraction
+    with ZipFileWithProgress(zip_file_path, "r") as zip_ref:
+        zip_ref.extract_zip(f"  Extracting  {name}", binary_path)  # type: ignore
+    print("")
+
+    # Clean up zip
+    print_progress(f"  Cleaning up {name}", 0)
```

Comment:
please use a logger instead of print statements in lines 117,118,123,126

Reply:
I replaced one of the calls but I want a progress bar and logger can't do that